src/components/Modals/AddLiquidityModal.js

```diff
@@ -87,55 +78,41 @@ function AddLiquidityModal({
   };
 
   const validate = (v, assetBalance, decimals, asset) => {
-    if (Number.isNaN(parseInt(v))) {
+    if (Number.isNaN(Number(v))) {
       return 'Not a valid number';
     }
-    const decimalsBN = getDecimalsBN(asset, decimals);
-    const inputBN = calculateAmountDesiredFormatted(v, decimalsBN);
+    const sanitizedValue = sanitizeValue(v.toString());
+    const inputBN = parseProperlyValue(asset, sanitizedValue, decimals);
     const assetBN = new BN(assetBalance);
     if (inputBN.gt(assetBN)) {
       return 'Input greater than balance';
     }
     return true;
   };
+  const { enum1, enum2 } = convertToEnumDex(asset1, asset2);
 
-  const handleChangeInput = (e, asset) => {
-    if (!reservesThisAssets) return;
+  const handleChangeInput = async (e, asset) => {
     const { value } = e.target;
-    let rate = null;
-
-    const isAsset1 = asset === reservesThisAssets.asset1Number;
 
-    if (!isAsset1) {
-      rate = getExchangeRate(
-        reservesThisAssets.asset1,
-        reservesThisAssets.asset2,
-        assetData1?.decimals,
-        assetData2?.decimals,
-      );
-    } else {
-      rate = getExchangeRate(
-        reservesThisAssets.asset2,
-        reservesThisAssets.asset1,
-        assetData2?.decimals,
-        assetData1?.decimals,
-      );
-    }
-    const exchangeOtherAssetValue = value * rate;
-
-    if (isAsset1) {
-      setValue('amount1Desired', value);
-      setValue('amount2Desired', exchangeOtherAssetValue);
-    } else {
-      setValue('amount1Desired', exchangeOtherAssetValue);
-      setValue('amount2Desired', value);
-    }
+    const isAsset1 = asset === asset1;
+    const sanitizedValue = sanitizeValue(value.toString());
+    const assetFormat = isAsset1 ? asset2 : asset1;
+    const amount = parseProperlyValue(
+      isAsset1 ? asset1 : asset2,
+      sanitizedValue,
+      isAsset1 ? assetData1?.decimals : assetData2?.decimals,
+    );
+    const tradeData = await (isAsset1
+      ? getSwapPriceExactTokensForTokens
+      : getSwapPriceTokensForExactTokens)(enum1, enum2, amount, false);
```

Comment:
```suggestion
    const getSwapPrice = isAsset1 ? getSwapPriceExactTokensForTokens : getSwapPriceTokensForExactTokens;
    const tradeData = await getSwapPrice(enum1, enum2, amount, false);
```
Will be much more readable.

Reply:
The same pattern is repeated in multiple places btw.